### DISCUSSION TITLE
Angle between a scratch and mean orientation of the grain where the scratch resides

### DISCUSSION THREAD

**Discussion by Mustafa-Rifat** [mtex-toolbox/mtex#727] [Ask Anything] [answered]
Hello mtex community,

The EBSD plot that I have shown here have 6 scratches, made intentionally. Let's consider the 3rd scratch from right. The green arrow shows the scratch direction. The scratch passes through 3 grains. I want to find the angle between the mean orientation of each of the grains with the scratch direction (i.e. angle between the mean orientation of grain 1 and scratch direction; angle between the mean orientation of grain 2 and scratch direction;angle between the mean orientation of grain 3 and scratch direction).
I know how to get the mean orientation of a grain. I use:
`grains (x,y).meanOrientation`
Here x and y are any point coordinate which is inside a grain. This gives me euler angle. If I find the mean orientation of another grain, I can use this line of code to find the misorientation between 2 grains using this code:
`angle(grains (p1(1),p1(2)).meanOrientation,grains (a,b).meanOrientation)/degree  `
Where (a,b) is a point coordinate inside the second grain. Can anyone help me how to find the angle between the scratch and mean orientation of a grain here?
![1](https://user-images.githubusercontent.com/60717121/105912811-0416d500-5ffa-11eb-8cac-3e14fb49a379.jpg)

**Comment by kilir** [comment, marked as answer]
Hi,
according to the title you are looking for the angle between a direction and an orientation. I don't think this can be defined. Can you give an example what this should be?

Asking for the direction between a direction ("scratch") and an orientation is equally confusing to me. Maybe you mean the crystal direction parallel to the scratch? Maybe you need to explain a little more in detail.

Cheers,
Rüdiger

**Reply by Mustafa-Rifat** [reply]
Hello Rüdiger,

I made a mistake in the last question. What I meant is 'how to find the angle between the scratch and mean orientation of a grain here?' I edited it and apologies for that.
What I want actually is the scratch direction with respect to the crystal direction.
1. Is it possible to find the angle between a crystal direction, let's say [100] and the scratch direction here?
2. Also, is it possible to find what you just suggested i.e. find the crystal direction which is parallel to the scratch direction.

Thanks a lot!

**Reply by kilir** [reply]
Hi Mustafa,
orientation .* crystal direction(s) will give you the direction(s) in specimen coordinates and inv(orientation).* specimen direction ref will give you the crystal direction parallel to ref. Then you can use `angle`.
See also here: https://mtex-toolbox.github.io/DefinitionAsCoordinateTransform.html

Hope that helps.

Cheers,
Rüdiger

**Reply by Mustafa-Rifat** [reply]
Let's say I am considering `r = ori * h` from the link: [https://mtex-toolbox.github.io/DefinitionAsCoordinateTransform.html](url)
These are the operations I am performing:
```
%defining the crystal symmetry similar to the way I defined in the EBSD plot
cs = crystalSymmetry('m-3m',[3.57 3.57 3.57],'mineral', 'Ni-superalloy', 'color', 'light blue')
%defining the [100] direction
h = Miller(1,0,0,cs,'uvw')
%finding the mean orientation of the grain I am interested
ori=grains(151,25).meanOrientation
%transferring h to specimen coordinate
r=ori*h
%finding the angle
angle(r,vector3d.Y,'noSymmetry') / degree
```
I have got an answer about 65 degrees.
However, You will see that in I have taken vector3d.Y at last line with r. This is (i.e. [0 1 0]) what I assumed to be the direction of the scratch in the specimen. Is it alright?

**Reply by kilir** [reply]
Hi Mustafa,
`vector3d.Y` is the same as `vector3d(0,1,0)`, however, whether this direction coincides with the direction of your scratch as you see it on the screen depends on how you set the plotting conventions. Maybe just use `xy = ginput(2)` to get the start and end coordinates of your scratch when in doubt where your x-axis points to.

Another note: you may want to use `r=ori*h.symmetrise('unique')` since I guess you can't distinguish between symmetrically equivalent crystal directions.

Cheers,
Rüdiger

**Reply by Mustafa-Rifat** [reply]
Hello Rüdiger,

Thanks a lot for your quick replies. I did find the start and end point of the arrow and saw there is an increase of y-coordinate. So, I have chosen the vector3d.Y.

When I use
```
 r=ori*h.symmetrise('unique')
angle(r,vector3d.Y,'noSymmetry') / degree
```
 as you suggested, I get 6 angles. Moreover, I found `h.symmetrise ` to be 
  ```
u 1  0  0 -1  0  0
  v 0  1  0  0 -1  0
  w 0  0  1  0  0 -1
```
So, I guess the 6 angles that I get are the angles between scratch direction [0 1 0] and the 6 [u v w] here respectively, right?

**Reply by kilir** [reply]
Exactly.
Adding `'antipodal'` should leave you with three unique values.
Rüdiger
